File: src/python/transaction/operations/account_update/iterate_roles.py
import asyncio

from wax import create_hive_chain
from wax.complex_operations.account_update import AccountAuthorityUpdateOperation


async def main() -> None:
    # Initialize hive chain interface
    wax = create_hive_chain()

    # Initialize a transaction object
    tx = await wax.create_transaction()

    # Create online operation - createFor will parse authorities for "gtg" account from the chain
    op = await AccountAuthorityUpdateOperation.create_for(wax, account_name="gtg")

    # Iterate over all role levels of hive role category
    for role in op.categories.hive:
        # Print the current role value
        print(str(role.value))

        # Warn if role is null authority
        if role.level != "memo" and role.is_null_authority:
            print("Role is null authority")

    # We do not have to push operation to the transaction as we just wanted to iterate over roles


asyncio.run(main())

File: src/rust/transaction/operations/account_update/iterate_roles.rs
use wax::prelude::*;
use wax::complex_operations::AccountAuthorityUpdateOperation;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize hive chain interface
    let chain = create_hive_chain(None)?;

    // Initialize a transaction object
    let _tx = chain.create_transaction(None).await?;

    // Create online operation - create_for will parse authorities for "gtg" account from the chain
    let op = AccountAuthorityUpdateOperation::create_for(&chain, "gtg").await?;

    // Iterate over all authority role levels of the hive role category
    for role in [&op.owner, &op.active, &op.posting] {
        // Print the current role value
        println!("{:#?}", role.value());

        // Warn if role is null authority
        if role.is_null_authority() {
            eprintln!("Role is null authority");
        }
    }

    // Print the current memo key
    println!("{}", op.memo.value());

    // We do not have to push operation to the transaction as we just wanted to iterate over roles

    Ok(())
}
